presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraPartitionManager.java

```diff
@@ -196,4 +202,48 @@ public CassandraPartitionResult getPartitions(ConnectorTableHandle tableHandle,
         }
         return Sets.cartesianProduct(partitionColumnValues.build());
     }
+
+    // todo should this be a complete copy of getPartitionKeysSet? for ex., why do you require that the range be a single value"?
```

Comment:
See this: http://www.datastax.com/dev/blog/a-deep-look-to-the-cql-where-clause
The restrictions are different; for partitioning keys, it has to be a single value. For clustering keys, the last one can be a range, but the rest need to be single values.